From d338431dcb9d19cf717776a85ef14d7c9fa452f9 Mon Sep 17 00:00:00 2001
From: Kit Sczudlo <kit@kitscz.com>
Date: Tue, 17 Sep 2024 12:30:40 -0700
Subject: [PATCH] Add in accomodations for max9867 codec quirks

---
 components/esp_driver_i2s/i2s_std.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/components/esp_driver_i2s/i2s_std.c b/components/esp_driver_i2s/i2s_std.c
index c3ef0c0469..3d0ab2715f 100644
--- a/components/esp_driver_i2s/i2s_std.c
+++ b/components/esp_driver_i2s/i2s_std.c
@@ -38,6 +38,14 @@ static esp_err_t i2s_std_calculate_clock(i2s_chan_handle_t handle, const i2s_std
     if (handle->role == I2S_ROLE_MASTER) {
         clk_info->bclk = rate * handle->total_slot * slot_bits;
         clk_info->mclk = rate * clk_cfg->mclk_multiple;
+
+        // XXX The MAX9867 requires an MCLK of 10MHz or above - force that here
+        if (clk_info->mclk < 5000000) {
+            clk_info->mclk =  rate * clk_cfg->mclk_multiple * 4;
+        } else if (clk_info->mclk < 10000000) {
+            clk_info->mclk = rate * clk_cfg->mclk_multiple * 2;
+        }
+
         clk_info->bclk_div = clk_info->mclk / clk_info->bclk;
     } else {
         /* For slave mode, mclk >= bclk * 8, so fix bclk_div to 2 first */
-- 
2.46.0

