From 8d38e8de42ada3cf34d2a7e7da000a3cf07641d1 Mon Sep 17 00:00:00 2001
From: Kit Sczudlo <kit@kitscz.com>
Date: Tue, 17 Sep 2024 12:25:20 -0700
Subject: [PATCH] Force menuconfig option to impact i2c_bus driver

---
 components/esp_peripherals/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/esp_peripherals/CMakeLists.txt b/components/esp_peripherals/CMakeLists.txt
index ab27f35d..70103a6a 100644
--- a/components/esp_peripherals/CMakeLists.txt
+++ b/components/esp_peripherals/CMakeLists.txt
@@ -30,7 +30,7 @@ list(APPEND COMPONENT_SRCS ./esp_peripherals.c
                 ./lib/tca9554/tca9554.c
                 ./lib/gpio_isr/gpio_isr.c)
 
-if("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_LESS "5.3")
+if((CONFIG_CODEC_I2C_BACKWARD_COMPATIBLE STREQUAL "y") OR ("${IDF_VERSION_MAJOR}.${IDF_VERSION_MINOR}" VERSION_LESS "5.3"))
     list(APPEND COMPONENT_SRCS  ./driver/i2c_bus/i2c_bus.c)
 else()
     list(APPEND COMPONENT_SRCS  ./driver/i2c_bus/i2c_bus_v2.c)
-- 
2.46.0

