From 68a3ed6bd44b947189178d9dbe1e133cb694fa91 Mon Sep 17 00:00:00 2001
From: kscz <kit@kitscz.com>
Date: Mon, 1 Apr 2024 11:27:54 -0700
Subject: [PATCH] Add in patches to make i2s work with MAX9867

---
 components/driver/deprecated/i2s_legacy.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/components/driver/deprecated/i2s_legacy.c b/components/driver/deprecated/i2s_legacy.c
index 8159f8a4f8..a93fdbca3a 100644
--- a/components/driver/deprecated/i2s_legacy.c
+++ b/components/driver/deprecated/i2s_legacy.c
@@ -760,6 +760,14 @@ static esp_err_t i2s_calculate_common_clock(int i2s_num, i2s_hal_clock_info_t *c
     if (p_i2s[i2s_num]->role == I2S_ROLE_MASTER) {
         clk_info->bclk = rate * slot_num * slot_bits;
         clk_info->mclk = rate * clk_cfg->mclk_multiple;
+
+        // XXX The MAX9867 requires an MCLK of 10MHz or above - force that here
+        if (clk_info->mclk < 5000000) {
+            clk_info->mclk =  rate * clk_cfg->mclk_multiple * 4;
+        } else if (clk_info->mclk < 10000000) {
+            clk_info->mclk = rate * clk_cfg->mclk_multiple * 2;
+        }
+
         clk_info->bclk_div = clk_info->mclk / clk_info->bclk;
     } else {
         /* For slave mode, mclk >= bclk * 8, so fix bclk_div to 8 first */
-- 
2.44.0

