diff --git a/components/esp_driver_i2s/i2s_std.c b/components/esp_driver_i2s/i2s_std.c
index 32f4e4125cc..10e847275aa 100644
--- a/components/esp_driver_i2s/i2s_std.c
+++ b/components/esp_driver_i2s/i2s_std.c
@@ -39,6 +39,14 @@ static esp_err_t i2s_std_calculate_clock(i2s_chan_handle_t handle, const i2s_std
     if (handle->role == I2S_ROLE_MASTER || handle->full_duplex_slave) {
         clk_info->bclk = rate * handle->total_slot * slot_bits;
         clk_info->mclk = rate * clk_cfg->mclk_multiple;
+
+        // XXX The MAX9867 requires an MCLK of 10MHz or above - force that here
+        if (clk_info->mclk < 5000000) {
+            clk_info->mclk =  rate * clk_cfg->mclk_multiple * 4;
+        } else if (clk_info->mclk < 10000000) {
+            clk_info->mclk = rate * clk_cfg->mclk_multiple * 2;
+        }
+
         clk_info->bclk_div = clk_info->mclk / clk_info->bclk;
         if (clk_info->mclk % clk_info->bclk != 0) {
             ESP_LOGW(TAG, "the current mclk multiple cannot perform integer division (slot_num: %"PRIu32", slot_bits: %"PRIu32")", handle->total_slot, slot_bits);
